charmcraft/utils.py

```diff
@@ -136,3 +140,52 @@ def useful_filepath(filepath):
     if not filepath.is_file():
         raise CommandError("{!r} is not a file.".format(str(filepath)))
     return filepath
+
+
+def get_os_platform(filepath=pathlib.Path("/etc/os-release")):
+    """Determine a system/release combo for an OS using /etc/os-release if available."""
+    system = platform.system()
+    release = platform.release()
+    machine = platform.machine()
+
+    if system == "Linux":
+        os_release = {}
+        try:
+            with filepath.open("r", encoding='utf-8') as f:
+                for line in f:
+                    if "=" in line:
+                        key, value = line.rstrip().split("=", 1)
+                        os_release[key] = value.strip('"')
```

Comment:
Seems unlikely, but if `"` were to show up in a string, this would break it.

```suggestion
                        value = value.strip()
                        if value.startswith('"') and value.endswith('"'):
                            value = value[1:-1]
                        os_release[key] = value
```

Reply:
Nice, I added several tests for this case and similar.